crates/mun_memory/src/gc/mark_sweep.rs

```diff
@@ -446,7 +455,7 @@ where
     type Array = ArrayHandle;
 
     fn alloc(&self, ty: &Type) -> GcPtr {
-        let object = alloc_obj(ty.clone());
+        let object = alloc_struct(ty.clone());
```

Comment:
Couldnt a type here also be a primitive? Any concrete type could be allocated via the GC.